service/internal/changedetection/tccpn.go

```diff
@@ -60,10 +68,39 @@ func (t *TCCPN) ShouldUpdate(ctx context.Context, cr infrastructurev1alpha2.AWSC
 		}
 	}
 
+	var currentRelease releasev1alpha1.Release
+	{
+		currentRelease, err = t.releases.Release(ctx, cc.Status.TenantCluster.ReleaseVersion)
+		if releases.IsNotFound(err) {
+			// fall through
+		} else if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
+
+	var targetRelease releasev1alpha1.Release
+	{
+		targetRelease, err = t.releases.Release(ctx, key.ReleaseVersion(&cr))
+		if releases.IsNotFound(err) {
+			// fall through
+		} else if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
+
+	componentVersionsEqual := releaseComponentsEqual(currentRelease, targetRelease)
 	masterInstanceEqual := cc.Status.TenantCluster.TCCPN.InstanceType == key.ControlPlaneInstanceType(cr)
 	masterReplicasEqual := cc.Status.TenantCluster.TCCPN.MasterReplicas == rep
 	operatorVersionEqual := cc.Status.TenantCluster.OperatorVersion == key.OperatorVersion(&cr)
 
+	if !componentVersionsEqual {
+		t.logger.LogCtx(ctx,
+			"level", "debug",
+			"message", "detected TCCPN stack should update",
+			"reason", fmt.Sprintf("component versions changed: %v", componentsDiff(currentRelease, targetRelease)),
```

Comment:
Super nitpick but I hate colons in logs because we usually have a JSON log formatter. What I would like to see is something like this in natural language without additional structural syntax. 

```
component version changed from X to Y
```

Reply:
Changed the format of reason, this will print out components diff like `etcd version changed from 3.4.9 to 3.4.10, kubernetes version changed from 1.16.13 to 1.17.9`